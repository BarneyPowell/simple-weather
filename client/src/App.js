import React, { Component } from 'react';

import DailyWeatherContainer from './containers/DailyWeatherContainer'

import { WeatherApiProvider } from './providers/WeatherApiProvider';

class App extends Component {



  render() {

    return (
      <WeatherApiProvider>
        <div className="App">
          <DailyWeatherContainer className="main-weather" />
        </div>
      </WeatherApiProvider>
    );
  }
}

export default App;
