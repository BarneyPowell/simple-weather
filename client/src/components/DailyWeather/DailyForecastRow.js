import React from 'react';

import DateCell from './DateCell';
import HourlyForecastCell from './HourlyForecastCell';

export default (props) => {
    const { day } = props;
    return (
        <li >
        <DateCell date={day.day} />
        <ol className="daily-weather_hours">
            {day.hours.map((hourly, idx) => 
                <HourlyForecastCell key={`${hourly.time}`} 
                    temperature={hourly.temp} 
                    time={hourly.time}
                    weather={hourly.weather} />
            )}
        </ol>
    </li>
    );
}