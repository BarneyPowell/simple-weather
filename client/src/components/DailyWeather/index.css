
.daily-weather {
    position: relative;
}

.daily-weather.-empty {
    text-align: center;
}

.daily-weather .-loading {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(255,255,255,0.8);
    display: flex;
    align-items:  center;
    justify-content: center;
}

.daily-weather .-loading > span {
    flex: 0 1 auto;
    
}

.daily-weather ol {
    list-style: none;
    margin: 0;
    padding: 0;
}


.daily-weather_days {
    display: flex;
    flex-direction: column;
}

.daily-weather_days > li {
    display: flex;
    flex-direction: row;
    flex: 0 1 auto;
    border-bottom: 1px solid #abcdef;
}

.daily-weather_days > li:first-child {
    border-top: 1px solid #abcdef;
}


.daily-weather_day {
    width: 10%;
    flex: 0 1 auto;

    background-color: #ddeeff;

    height: 4em;
    width: 4em;

    display: flex;
    flex-direction: column;
}

.daily-weather_day .-date {
    font-size: 0.625em;

    flex: 0 1 auto;
    text-align: center;
    padding: 0.2em;
}

.daily-weather_day .-day {
    flex: 0 1 auto;
    height: 100%;
    font-size: 1.4em;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
}

.daily-weather_day .-day span {
    flex: 0 1 auto;
}

.daily-weather_hours {
    display: flex;
    width: 100%;
    flex: 0 1 auto;
    overflow: auto;
    justify-content: space-between;
}



.hourly-forecast {
    flex: 0 0 auto;
    min-width: 4em;
    height: 4em;
    position: relative;
    display: flex;
    flex-direction: column;
}


.hourly-forecast .-temperature {
    flex: 0 1 auto;
    display: flex;
    height: 100%;
    align-items: center;
    justify-content: center;
}

.hourly-forecast .-temperature span {
    flex: 0 1 auto;
}

.hourly-forecast .-weather {
    flex: 0 1 auto;
    display: flex;
    height: 100%;
    align-items: center;
    justify-content: center;
    font-size: 0.6em;
}

.hourly-forecast .-weather span {
    flex: 0 1 auto;
}


.hourly-forecast .-hour {
    flex: 0 1 auto;
    text-align: center;
    font-size: 0.625em;
    padding: 0.2em;
}

.daily-weather_header {
    display: flex;
    flex-direction: column;
    padding: 1em;
}

.daily-weather_header .-intro {
    font-size: 0.7em;
    flex: 0 1 auto;
}

.daily-weather_header .-location {
    font-size: 1.5em;
    font-weight: bold;
    flex: 0 1 auto;
}