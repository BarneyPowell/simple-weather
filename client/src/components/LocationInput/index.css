.location-input {
    display: flex;
    flex-direction: column;
    padding: 1em;
}

.location-input label {
    flex: 0 1 auto;
    margin-bottom: 0.5em;
    font-size: 0.8em;
}

.location-input input {
    flex: 0 1 auto;
    border: none;
    border-bottom: 2px solid #abcdef;
    background-color: #f9f9f9;
    padding: 0.5em;
    font-size: 1em;
}