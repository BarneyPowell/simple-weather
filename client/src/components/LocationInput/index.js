import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.css';

const KeyCodes = {
    ENTER: 13
}

class LocationInput extends Component
{

    constructor(props) {
        super(props);

        this.handleKeyPress = this.handleKeyPress.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleKeyPress(e) {
        if(e.which === KeyCodes.ENTER
            && this.props.onAction
            && typeof this.props.onAction === 'function') 
            {
                this.props.onAction(e.target.value);
            }
    }

    handleChange(e) {
        if(this.props.onChange && typeof this.props.onChange === 'function') {
            this.props.onChange(e.target.value);
        }
    }

    render() {
        return (
            <div className="location-input">
                <label>{this.props.label}</label>
                <input type="text" 
                    onKeyPress={this.handleKeyPress} 
                    onChange={this.handleChange}
                    placeholder="City name" />
            </div>
        );
    }
}

LocationInput.propTypes = {
    label: PropTypes.string,
    onChange: PropTypes.func,
    onAction: PropTypes.func
}

export default LocationInput;