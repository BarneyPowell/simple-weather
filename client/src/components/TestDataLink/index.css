.test-data-link {
    text-align: center;
    font-size: 0.8em;
    margin-bottom: 1em;
}

.test-data-link button {
    margin: 0;
    padding: 0;
    font-size: 1em;
    text-decoration: underline;
    color: #789abc;
}