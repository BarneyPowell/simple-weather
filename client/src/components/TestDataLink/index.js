import React from 'react';
import './index.css';

export default (props) => {
    return (
        <div className="test-data-link">
            or <button onClick={props.onClick}>use test data from csv</button>
        </div>
    )
}