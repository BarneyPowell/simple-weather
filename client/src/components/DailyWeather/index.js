import React, { Component } from 'react';
import PropTypes from 'prop-types';

import HourlyForecastCell from './HourlyForecastCell';
import DateCell from './DateCell';
import Header from './Header';
import DailyForecastRow from './DailyForecastRow';
import './index.css';


class DailyWeather extends Component
{


    renderEmpty(classNames) {
        classNames.push('-empty');
        return (
            <div className={classNames.join(' ')}>
                <span>No forecast found</span>
                {this.renderLoadingMessage()}
            </div>
        );
    }

    renderLoadingMessage() {
        if(!this.props.isFetching) return null;
        return (
            <div className="-loading">
                <span>Fetching forecast...</span>
            </div>
        );
    }

    render() {

        const classNames = [
            this.props.className
        ];

        classNames.push('daily-weather');

        const forecasts = this.props.forecasts || [];

        if(this.props.isFetching) classNames.push('-isFetching');

        if(forecasts.length === 0) {
            return this.renderEmpty(classNames);
        }
        
        return (
            <div className={classNames.join(' ')}>
                <Header location={this.props.location} />
                <ol className="daily-weather_days">
                    {forecasts.map((day, idx) => <DailyForecastRow key={`${day.day}`} day={day} />)}
                </ol>
                {this.renderLoadingMessage()}
            </div>
        );
    }
}

DailyWeather.propTypes = {
    location: PropTypes.object,
    forecasts: PropTypes.array,
    isFetching: PropTypes.bool
}

export default DailyWeather;