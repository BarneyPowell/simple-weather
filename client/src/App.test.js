import React from 'react';
import App from './App';
import DailyWeather from './components/DailyWeather';
import LocationInput from './components/LocationInput';
import TestDataLink from './components/TestDataLink';

it('renders without crashing', () => {
  const wrapper = mount(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('renders a DailyWeather component', () => {
  const wrapper = mount(<App />);
  expect(wrapper.find(DailyWeather)).toHaveLength(1);
});

  it('renders a link to enable test data', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(TestDataLink)).toHaveLength(1);
});

it('renders an input box to collect location input', () => {
  const wrapper = mount(<App />);
  expect(wrapper.find(TestDataLink)).toHaveLength(1);
});