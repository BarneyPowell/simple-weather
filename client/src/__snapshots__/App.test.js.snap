// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<App>
  <WeatherApiProvider>
    <div>
      <div
        className="App"
      >
        <Wrapper
          className="main-weather"
        >
          <DailyWeatherContainer
            className="main-weather"
            fetchForecast={[Function]}
            getForecast={[Function]}
            setUseTestData={[Function]}
          >
            <div>
              <LocationInput
                label="Enter a city, and press enter"
                onAction={[Function]}
              >
                <div
                  className="location-input"
                >
                  <label>
                    Enter a city, and press enter
                  </label>
                  <input
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="City name"
                    type="text"
                  />
                </div>
              </LocationInput>
              <Component
                onClick={[Function]}
              >
                <div
                  className="test-data-link"
                >
                  or 
                  <button
                    onClick={[Function]}
                  >
                    use test data from csv
                  </button>
                </div>
              </Component>
              <DailyWeather
                className="main-weather"
                forecasts={Array []}
                isFetching={false}
                shouldUseTestData={false}
              >
                <div
                  className="main-weather daily-weather -empty"
                >
                  <span>
                    No forecast found
                  </span>
                </div>
              </DailyWeather>
            </div>
          </DailyWeatherContainer>
        </Wrapper>
      </div>
    </div>
  </WeatherApiProvider>
</App>
`;
